lock-api/src/main/java/com/palantir/lock/LockDescriptor.java

```diff
@@ -48,25 +48,30 @@
     }
 
     @JsonIgnore
+    @SuppressWarnings("checkstyle:RegexpSinglelineJava") // JDK StandardCharsets is only a feature in 1.7+
```

Comment:
As mentioned later - my main concern is that clients of large internal product may mandate Java 6 (need to check if this is still the case). I've filed #1875 to track this separately.